### ISSUE TITLE
Bug: `no-useless-key` doesn't report on TemplateLiterals

### ISSUE THREAD

**Issue by abrahamguo** [eslint/eslint#19280]
### Environment

Environment Info:

Node version: v22.12.0
npm version: v11.0.0
Local ESLint version: v9.17.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.1.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
/* eslint no-useless-computed-key: "error" */
```
</details>

```js
const foo = { [`a`]: 1 };
```

### What did you expect to happen?

The useless computed key should be reported.

### What actually happened?

It was not reported.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVzZWxlc3MtY29tcHV0ZWQta2V5OiBcImVycm9yXCIgKi9cbmNvbnN0IGZvbyA9IHsgW2BhYF06IDEgfTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
This is the intentional behavior of this rule. It reports only when `[]` can be just removed. In this case, ``const foo = { `a`: 1 };`` would be invalid syntax.
